960-api-services/src/main/java/io/harness/k8s/KubernetesContainerServiceImpl.java

```
@@ -1414,6 +1415,30 @@ private Object getVirtualServiceUsingK8sClient(
     return virtualServiceSupplier.get();
   }
 
+  @Override
+  public Object patchCustomObject(
+      KubernetesConfig kubernetesConfig, String name, K8sApiVersion apiVersion, String plural, String patch) {
+    if (kubernetesConfig == null) {
+      return null;
+    }
+    final Supplier<Object> customObjectSupplier = Retry.decorateSupplier(retry, () -> {
+      try {
+        ApiClient apiClient = kubernetesHelperService.getApiClientWithReadTimeout(kubernetesConfig);
+        CustomObjectsApi customObjectsApi = new CustomObjectsApi(apiClient);
+        return customObjectsApi.patchNamespacedCustomObject(apiVersion.getGroup(), apiVersion.getVersion(),
+            kubernetesConfig.getNamespace(), plural, name, new V1Patch(patch), null, null, null);
+      } catch (ApiException exception) {
+        String message = format(
+            "Unable to patch custom object: Group:[%s] ApiVersion:[%s] Namespace:[%s], Name:[%s] Code: %s, message: %s",
+            apiVersion.getGroup(), apiVersion.getVersion(), kubernetesConfig.getNamespace(), name, exception.getCode(),
+            getErrorMessage(exception));
+        log.error(message);
```

just curious, we'll just log our failure to patch and continue with the overall execution?

---

I feel like that's the right thing to do, as at this point we already swapped service selectors, and I don't see the point of failing the whole step if updating the destination fails. We just print a warning and the customer may update it by themself if patching fails. 
This is just a suggestion, if you think that we should fail it we can discuss it?

---

It should fail, from the deployment perspective we weren't able to bring it to desired state so user should be aware of it

---

added error handling (execution will fail with HintException) if this method returns null

---

I don't think this is a good idea, we're losing the context of why it failed, it's better to propagate the error and if you want to add more details just try catch it

---

done